app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :email, presence: true, length: { minium: Settings.min_leng, maximum: Settings.max_leng },
+                    format: { :with => /Settings.regex_email/ },
+                    uniqueness: { case_sensitive: false }
+
+  validates :name, presence: true, length: { maximum: Settings.max_leng }
+  validates :password, presence: true, length: { minimum: Settings.min_pass }, if: :password
+  has_secure_password
+
+  private
+
+  def downcase_email
+    self.email.downcase!
```

Comment:
Có cần self ko em

Reply:
bỏ self đi được nè

Reply:
e thấy self nó trỏ đến đối tượng hiện tại đang sử dụng, khá giống với con trỏ this trong các ngôn ngữ khác, nhưng e thấy nó vẫn khác với this. ví dụ self nằm trong instance method thì nó là object, còn trong class method là class